scripts/test/Muon/utilities/muon_workspace_test.py

```diff
@@ -0,0 +1,387 @@
+    table.addColumn("int", "col1", 0)
+    table.addColumn("int", "col2", 0)
+    [table.addRow([i + 1, 2 * i]) for i in range(4)]
+    return table
+
+
+class MuonWorkspaceTest(unittest.TestCase):
+    """
+    The MuonWorkspace object is a key class in the muon interface. It is a wrapper around a normal
+    Mantid workspace, which maintains a handle to the workspace whilst allowing it to be in the ADS
+    or not.
+
+    This allows certain workspaces to be held in the interface, and displayed to the user as and
+    when they are necessary.
+
+    It has some extra functionality which allows the workspace to be put in the ADS inside a "folder
+    structure" using workspace groups. So for example if I give the name "dir1/dir2/name" then a
+    workspace called "name" will be placed inside a group called "dir2" which will istelf be placed
+    inside a group called "dir1".
+
+    This allows the complex array of different workspaces from the muon interface to be structured
+    in the ADS to improve the user experience.
+
+    """
+
+    def setUp(self):
+        self.workspace = create_simple_workspace(data_x=[1, 2, 3, 4], data_y=[10, 10, 10, 10])
+        simpleapi.mtd.clear()
+
+        assert isinstance(self.workspace, Workspace2D)
+        assert simpleapi.mtd.size() == 0
+
+    def tearDown(self):
+        # clear the ADS
+        simpleapi.mtd.clear()
+
+    # ----------------------------------------------------------------------------------------------
+    # Test Initialization
+    # ----------------------------------------------------------------------------------------------
+
+    def test_that_cannot_initialize_without_supplying_a_workspace(self):
+        with self.assertRaises(TypeError):
+            MuonWorkspace()
+
+    def test_that_can_initialize_with_Workspace2D_object(self):
+        MuonWorkspace(workspace=self.workspace)
+
+    def test_that_can_initialize_with_TableWorkspace_object(self):
+        table_workspace = create_simple_table_workspace()
+        assert isinstance(table_workspace, ITableWorkspace)
+
+        MuonWorkspace(workspace=table_workspace)
+
+    def test_that_cannot_initialize_with_WorkspaceGroup_object(self):
+        group_workspace = api.WorkspaceGroup()
+        assert isinstance(group_workspace, WorkspaceGroup)
+
+        with self.assertRaises(AttributeError):
+            MuonWorkspace(workspace=group_workspace)
+
+    def test_that_cannot_initialize_with_non_workspace_objects(self):
+        with self.assertRaises(AttributeError):
+            MuonWorkspace(workspace="string")
+
+        with self.assertRaises(AttributeError):
+            MuonWorkspace(workspace=1234)
+
+        with self.assertRaises(AttributeError):
+            MuonWorkspace(workspace=5.5)
+
+    def test_that_initialized_object_is_not_in_ADS_by_default(self):
+        workspace_handle = MuonWorkspace(workspace=self.workspace)
+
+        self.assertEqual(workspace_handle.is_hidden, True)
```

Comment:
we should also have `assert simpleapi.mtd.size() == 0`, otherwise we are only testing that the class thinks that the ADS is empty.